### PR TITLE
sbcl: fix build on 10.6 i386; update to 2.2.5

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 10K549 x86_64
Xcode 3.2.6 10M2518

as `port install sbcl build_arch=i386`

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by catap** [comment]
@herbygillot, @mascguy I feel that patching that compiler to support 10.6 i386 is crazy idea because it quite frigile and I not only build it, but introduce a new bugs.

Here I move by another approach: go back to a bit older version but which works fine on i386 :)

File: lang/sbcl/Portfile
Comment (mascguy) [inline lang/sbcl/Portfile:44]: Even though we're only supporting Darwin for this port, it's still good practice to surround the version check in a `platform darwin {` block.

Comment (catap) [reply lang/sbcl/Portfile:44]: @mascguy indeed.

**Comment by mascguy** [comment]
My suggestion is to wrap the version check within `platform darwin {` blocks, as it makes the version check logic easier to read.

**Comment by catap** [comment]
> My suggestion is to wrap the version check within `platform darwin {` blocks, as it makes the version check logic easier to read.

can you suggest an example of code? I feel that I miss something.

**Comment by mascguy** [comment]
> > My suggestion is to wrap the version check within `platform darwin {` blocks, as it makes the version check logic easier to read.
> 
> can you suggest an example of code? I feel that I miss something.

> > My suggestion is to wrap the version check within `platform darwin {` blocks, as it makes the version check logic easier to read.
> 
> can you suggest an example of code? I feel that I miss something.

```
platform darwin {
    if { ${os.major} <= 10 } {
        ...etc...
    }
}
```

**Comment by catap** [comment]
@mascguy I see. But it requires to reformat the file to keep things alligned :)

**Comment by mascguy** [comment]
> @mascguy I see. But it requires to reformat the file to keep things alligned :)

No, you only need to include the version check logic in that block. The rest of the portfile doesn't change.

**Comment by catap** [comment]
@mascguy the best that I may do in formating is:

Screenshot 2023-06-03 at 23 27 18

**Comment by mascguy** [comment]
> @mascguy the best that I may do in formating is:
> 
> Screenshot 2023-06-03 at 23 27 18
> 

Don't worry about trying to line up the checksum with the previous block. Simply indent appropriately.

**Comment by catap** [comment]
@barracuda156 may I ask you to try build 1.3.9 with threads and fancy on PPC? Maybe pin the PPC version also on that?

**Comment by catap** [comment]
From another hand, let me make one more attempt

**Comment by barracuda156** [comment]
> From another hand, let me make one more attempt, maybe I can find something newer for i386

@catap Please update on that, and yes, I will try then on PPC today.

At least something prior to moving to `libdispatch` would be nice. We may try asking upstream by the way, they are somewhat responsive, and perhaps can find `i386` hardware at least. If SDK problem is sorted, we can hopefully fix `ppc` too.

**Comment by catap** [comment]
> At least something prior to moving to `libdispatch` would be nice. We may try asking upstream by the way, they are somewhat responsive, and perhaps can find `i386` hardware at least. If SDK problem is sorted, we can hopefully fix `ppc` too.

Well.. It requires not only hardware, but also MacPorts infrastructure like Legacy Support or patching that seems like a major stop point to such work from upstream point of view.

**Comment by barracuda156** [comment]
@catap Okay, let us leave upstream aside for now. Which version should I try on PPC with threads (or with fancy)? 1.3.9 or something else?

**Comment by catap** [comment]
@barracuda156 let start from 1.3.9. It seems as good baseline.

**Comment by barracuda156** [comment]
@catap Fails pretty fast (when built with +fancy):
```
beginning GENESIS, creating headers in "src/runtime/genesis"
unhandled TYPE-ERROR:
  The value NIL is not of type SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION.

0: (SB-DEBUG::MAP-BACKTRACE
    #<CLOSURE (LAMBDA #) {1359817D}>
    :START
    0
    :COUNT
    128)
1: (SB-DEBUG:BACKTRACE 128 #<SYNONYM-STREAM :SYMBOL SB-SYS:*STDERR* {10129D11}>)
2: (SB-DEBUG::DEBUGGER-DISABLED-HOOK
    #<TYPE-ERROR
      expected-type: SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION datum: NIL>
    #<unavailable argument>)
3: (SB-DEBUG::RUN-HOOK
    SB-EXT:*INVOKE-DEBUGGER-HOOK*
    #<TYPE-ERROR
      expected-type: SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION datum: NIL>)
4: (INVOKE-DEBUGGER
    #<TYPE-ERROR
      expected-type: SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION datum: NIL>)
5: (ERROR
    TYPE-ERROR
    :DATUM
    NIL
    :EXPECTED-TYPE
    SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION)
6: (SB-KERNEL::OBJECT-NOT-TYPE-ERROR-HANDLER
    #<unavailable argument>
    #.(SB-SYS:INT-SAP #X01000750)
    #<SB-ALIEN-INTERNALS:ALIEN-VALUE :SAP #XBFFFE4D0 :TYPE (*
                                                            (SB-ALIEN:STRUCT
                                                             SB-VM::OS-CONTEXT-T-STRUCT))>
    (806 326))
7: (SB-KERNEL:INTERNAL-ERROR
    #.(SB-SYS:INT-SAP #XBFFFE4D0)
    #<unavailable argument>)
8: ("foreign function: call_into_lisp")
9: (SB!KERNEL::DSD-RAW-SLOT-DATA NIL)
10: (SB!KERNEL:DSD-RAW-TYPE NIL)
11: ((FLET SB!FASL::GET-SLOT-INDEX)
     (#<SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION SB!KERNEL::CLOS-HASH>
      #<SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION SB!KERNEL:CLASSOID>
      #<SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION SB!KERNEL::INVALID>
      #<SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION SB!KERNEL::INHERITS>
      #<SB!KERNEL:DEFSTRUCT-SLOT-DESCRIPTION SB!KERNEL::DEPTHOID> ...)
     #<SB!FASL::DESCRIPTOR for pointer: #X4100008, lowtag #b011, unknown>)
12: (SB!FASL::WRITE-SLOTS
     #<SB!FASL::DESCRIPTOR for pointer: #X10000F18, lowtag #b001, DYNAMIC>
     #<SB!KERNEL:LAYOUT for SB!KERNEL:LAYOUT {1258C0D1}>
     :INVALID
     #<SB!FASL::DESCRIPTOR for pointer: #X4100008, lowtag #b011, unknown>
     :INHERITS
     #<SB!FASL::DESCRIPTOR for pointer: #X10000F10, lowtag #b111, DYNAMIC>
     :DEPTHOID
     #<SB!FASL::DESCRIPTOR for fixnum: 0>
     :LENGTH
     #<SB!FASL::DESCRIPTOR for fixnum: 0>
     :INFO
     #<SB!FASL::DESCRIPTOR for pointer: #X4100008, lowtag #b011, unknown>
     :PURE
     #<SB!FASL::DESCRIPTOR for pointer: #X4100008, lowtag #b011, unknown>
     :BITMAP
     #<SB!FASL::DESCRIPTOR for fixnum: 0>
     :EQUALP-TESTS
     #<SB!FASL::DESCRIPTOR for pointer: #X10000F58, lowtag #b111, DYNAMIC>
     :SOURCE-LOCATION
     #<SB!FASL::DESCRIPTOR for pointer: #X4100008, lowtag #b011, unknown>
     :%FOR-STD-CLASS-B
     :SLOT-LIST
     #<SB!FASL::DESCRIPTOR for pointer: #X4100008, lowtag #b011, unknown>
     #<SB!FASL::DESCRIPTOR for fixnum: 0>)
13: (SB!FASL::MAKE-COLD-LAYOUT
     T
     #<SB!FASL::DESCRIPTOR for fixnum: 0>
     #<SB!FASL::DESCRIPTOR for pointer: #X10000F10, lowtag #b111, DYNAMIC>
     #<SB!FASL::DESCRIPTOR for fixnum: 0>
     #<SB!FASL::DESCRIPTOR for fixnum: 0>)
14: (SB!FASL::INITIALIZE-LAYOUTS)
15: (SB!VM:GENESIS
     :OBJECT-FILE-NAMES
     NIL
     :SYMBOL-TABLE-FILE-NAME
     NIL
     :CORE-FILE-NAME
     NIL
     :MAP-FILE-NAME
     NIL
     :C-HEADER-DIR-NAME
     "src/runtime/genesis")
16: (SB-INT:SIMPLE-EVAL-IN-LEXENV
     (SB!VM:GENESIS :C-HEADER-DIR-NAME "src/runtime/genesis")
     #<NULL-LEXENV>)
17: (EVAL (SB!VM:GENESIS :C-HEADER-DIR-NAME "src/runtime/genesis"))
18: (SB-EXT:INTERACTIVE-EVAL
     (SB!VM:GENESIS :C-HEADER-DIR-NAME "src/runtime/genesis")
     :EVAL
     NIL)
19: (SB-IMPL::REPL-FUN NIL)
20: ((LAMBDA ()))
21: (SB-IMPL::%WITH-REBOUND-IO-SYNTAX #<CLOSURE (LAMBDA #) {11E2D1DD}>)
22: (SB-IMPL::TOPLEVEL-REPL NIL)
23: (SB-IMPL::TOPLEVEL-INIT)
24: ((LABELS SB-IMPL::RESTART-LISP))

unhandled condition in --disable-debugger mode, quitting
Command failed:  cd "/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_lang_sbcl/sbcl/work/sbcl-1.3.9" && sh make.sh --prefix=/opt/local --xc-host=/opt/local/libexec/sbcl-bootstrap/bin/sbcl --arch=ppc --fancy 
Exit code: 1
Error: Failed to build sbcl: command execution failed
```

**Comment by catap** [comment]
@barracuda156 ok, let keep it for the next time.

**Comment by catap** [comment]
Status update: I do have 1.5.7 which works in 10.6 i386

**Comment by catap** [comment]
@mascguy I'm ready here! The second attempt was sucefful and I found the cause which brokes the stack and now it is compiles and works without downgrade of the version.

Thus, I've upgraded to the really the last version which supports i386 :)

**Comment by mascguy** [comment]
Looks good, but can you use `configure.build_arch` rather than `build_arch`?

**Review by mascguy — APPROVED** [review]
LGTM, will merge once CI completes

**Comment by catap** [comment]
Thus, it was surpassingly simple :)

**Comment by mascguy** [comment]
Oh, actually, we have to wait for @easye to approve. (Or until the 72-hour maintainer timeout, whichever comes first.)

**Comment by catap** [comment]
@mascguy I’m a co-maintainer of this port ;)

**Comment by mascguy** [comment]
> @mascguy I’m a co-maintainer of this port ;)

Ah, didn't notice that! Will merge then.

**Pull request merged**
